jac-uk/admin
# #2154 Configure working preferences questions

## What's included?
Additional configuration options for working preferences questions, as follows:

- [x] Enable multiple location questions. (e.g. this will enable preferred and maybe selections)
- [x] Enable multiple jurisdiction questions. (e.g. this will enable preferred and maybe selections)
- [x] Define how many answers candidates should select
- [x] Enable linked questions (e.g. if a particular answer is chosen then the user is asked a follow-on question)
- [x] Enable grouping of answers (two levels)
- [x] Allow re-use of jurisdictions from vacancy information (for jurisdiction questions)

Closes #2154 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?

We decided to continue to support current functionality for existing exercises and have new exercises benefit from the new functionality.

There are therefore two areas to test:

**1. There should not be any breaking changes for existing exercises**

Either pick an existing exercise that is configured with the original working preferences OR use the main [admin-develop](https://admin-develop.judicialappointments.digital/) url to create a new exercise and configure working preferences for it.

View the chosen/created exercise on the [Preview URL](https://jac-admin-develop--pr2207-feature-2154-d8s2h7zf.web.app/) and check that working preferences works as expected (as it does on admin-develop). For example you could have admin-develop and preview url running in separate windows and compare the functionality side by side.

You should be able to:

 - Configure and edit one location preference
 - Configure and edit one jurisdiction preference
 - Configure and edit multiple additional preferences
 - View a candidate's chosen preferences on their application
 - Edit a candidate's chosen preferences on their application (in 'edit mode')

_Note: you may like to create an application via [apply-develop](https://apply-develop.judicialappointments.digital/) so you can test viewing and editing applications._

Please highlight any breaking changes.


**2. New exercises should benefit from the new functionality and the new functionality satisfies user requirements**

Using the [Preview URL](https://jac-admin-develop--pr2207-feature-2154-d8s2h7zf.web.app/) please create a new exercise and check that the new working preferences functionality is available.

In particular check that the following functionality works as expected and also that the terminology and instructions provided are sufficiently clear.

1. Check that you can add multiple location questions
4. Check that you can add multiple jurisdiction questions
5. Check that you can define how many answers candidates should select (for multiple and ranked choice questions)
6. Check that you can create a linked / follow-on question (where if a user selects one answer they are asked a follow-on question)
7. Check that you can group answers
8. Check that, for jurisdiction questions, you can choose to use the exercise jurisdictions list instead of re-entering the data

_Note: when you have tested the above please test [this PR on Apply](https://github.com/jac-uk/apply/pull/1135) so that you have an application with the latest preferences data._

Then back in Admin continue to test the following in Applications:

9. Check that all working preferences answers are displayed correctly on the candidate's application
10. Check that all working preferences can be edited when the application is in 'edit mode'

## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
